main.go

```diff
@@ -163,65 +179,101 @@ func delete(cmd *cobra.Command, args []string) error {
+func getDbs() ([]string, error) {
+	filepath, err := getFilePath()
 	if err != nil {
-		return err
+		return nil, err
 	}
-	dbs, err := os.ReadDir(filepath.Join(dd, "kv"))
+	entries, err := os.ReadDir(filepath)
 	if err != nil {
-		return err
+		return nil, err
 	}
-	for _, d := range dbs {
-		if d.IsDir() {
-			fmt.Println("@" + d.Name())
+	var out []string
+	for _, e := range entries {
+		if e.IsDir() {
+			out = append(out, "@"+e.Name())
 		}
 	}
-	return nil
+	return out, nil
 }
 
-func deleteDb(cmd *cobra.Command, args []string) error {
-	n, err := nameFromArgs(args)
-	if err != nil {
-		return err
-	}
+// getFilePath: get the file path to the skate databases.
+func getFilePath(args ...string) (string, error) {
 	cc, err := client.NewClientWithDefaults()
 	if err != nil {
-		return err
+		return "", err
 	}
 	dd, err := cc.DataPath()
 	if err != nil {
-		return err
+		return "", err
 	}
-	dbs, err := os.ReadDir(filepath.Join(dd, "kv"))
+	return filepath.Join(append([]string{dd, "kv"}, args...)...), err
+}
+
+// deleteDb: delete a Skate database.
+func deleteDb(cmd *cobra.Command, args []string) error {
+	dbs, err := getDbs()
 	if err != nil {
 		return err
 	}
+	path, err := findDb(args[0], dbs)
+	var errNotFound errDBNotFound
+	if errors.As(err, &errNotFound) {
+		fmt.Printf("%q does not exist, %s\n", args[0], err.Error())
+		os.Exit(1)
+	}
+	if err != nil {
+		fmt.Printf("unexpected error: %s", err.Error())
+		os.Exit(1)
+	}
+	var confirmation string
+	fmt.Printf("are you sure you want to delete '%s' and all its contents?(y/n) ", warningStyle.Render(path))
+	fmt.Scanln(&confirmation)
+	if confirmation == "y" {
+		return os.RemoveAll(path)
+	}
+	fmt.Printf("did not delete %q\n", path)
+	return nil
+}
 
-	var found bool
-	for _, d := range dbs {
-		if d.IsDir() && d.Name() == n {
-			found = true
-			var confirmation string
-			fmt.Println(warningStyle.Render("are you sure you want to delete " + d.Name() + " and all its contents? (y/n)"))
-			fmt.Scanln(&confirmation)
-			if confirmation == "y" {
-				err := os.RemoveAll(filepath.Join(dd, "kv", d.Name()))
-				if err != nil {
-					return err
-				}
-			} else {
-				fmt.Println("did not delete " + d.Name())
+// findDb: returns the path to the named db or an errDBNotFound if no
+// match is found.
+func findDb(name string, dbs []string) (string, error) {
+	sName, err := nameFromArgs([]string{name})
+	if err != nil {
+		return "", err
+	}
+	path, err := getFilePath(sName)
+	if err != nil {
+		return "", err
+	}
+	_, err = os.Stat(path)
+	if sName == "" || os.IsNotExist(err) {
+		dbs, err := getDbs()
+		if err != nil {
+			return "", err
+		}
+		var suggestions []string
+		for _, db := range dbs {
+			levenshteinDistance := levenshtein.ComputeDistance(name, db)
+			suggestByLevenshtein := levenshteinDistance <= distance
+			suggestByPrefix := strings.HasPrefix(name, db[:distance])
+			suggestByPrefixAlt := strings.HasPrefix("@"+name, db[:distance])
```

Comment:
can users create dbs with 2 chars long names? if so, this will fail 🤔 

I'm also curious why both levenshtein and the prefix thing, did you see cases in which the prefix worked and levenshtein didn't?

Reply:
Awesome, just updated my tests to account for when a name might be longer than the available databases, if the name entered is a single character, and added an example db that's only two chars long. 

The distance is now compared to the difference in length of the input and db options.